cpp/test/sg/rf_test.cu

```diff
@@ -93,22 +143,41 @@ class RfTest : public ::testing::TestWithParam<RfInputs<T>> {
 
     // Inference data: same as train, but row major
     int inference_data_len = params.n_inference_rows * params.n_cols;
-    inference_data_h = {30.0, 10.0, 1.0, 20.0, 2.0, 10.0, 0.0, 40.0};
     inference_data_h.resize(inference_data_len);
+    for (int i = 0; i < params.n_inference_rows; i++) {
+      for (int j = 0; j < params.n_cols; j++) {
+        inference_data_h[i * params.n_cols + j] = data_h[j * params.n_rows + i];
+      }
+    }
 
     // Predict and compare against known labels
     predicted_labels.resize(params.n_inference_rows);
-    RF_metrics tmp = cross_validate(
-      handle, rf_classifier, inference_data_h.data(), labels_h.data(),
-      params.n_inference_rows, params.n_cols, predicted_labels.data(), false);
-    accuracy = tmp.accuracy;
+
+    if (!params.cross_validate) {
+      rf_classifier->predict(handle, inference_data_h.data(),
+                             params.n_inference_rows, params.n_cols,
+                             predicted_labels.data(), VERBOSE_TEST);
+
+      int num_correct;
+      for (int i = 0; i < params.n_inference_rows; i++) {
+        num_correct += (predicted_labels[i] == labels_h[i]);
+      }
+      accuracy = num_correct / params.n_inference_rows;
```

Comment:
The accuracy here will be floored to 0 as soon as num_correct < params.n_inference_rows (division of two ints). 
Also, num_correct should be initialized to 0.

Reply:
Thanks for the fix!